keras_rs/src/layers/embedding/base_distributed_embedding.py

```diff
@@ -24,6 +24,76 @@
 class DistributedEmbedding(keras.layers.Layer):
     """DistributedEmbedding, a layer for accelerated large embedding lookups.
 
+    ---
+
+    ## Note: `DistributedEmbedding` is in Preview.
+
+    ---
+
+    ## Configuration
+
+    A `DistributedEmbedding` embedding layer is configured via a set of
+    `keras_rs.layers.FeatureConfig` objects, which themselves refer to
+    `keras_rs.layers.TableConfig` objects.
+
+    - `TableConfig` defines an embedding table with parameters such as its
+      vocabulary size, embedding dimension, as well as a combiner for reduction
+      and optimizer for training.
+    - `FeatureConfig` defines what input features the `DistributedEmbedding`
+      will handle and which embedding table to use. Note that multiple features
+      can use the same embedding table.
+
+    ```python
+    table1 = keras_rs.layers.TableConfig(
+        name="table1",
+        vocabulary_size=TABLE1_VOCABULARY_SIZE,
+        embedding_dim=TABLE1_EMBEDDING_SIZE,
+    )
+    table2 = keras_rs.layers.TableConfig(
+        name="table2",
+        vocabulary_size=TABLE2_VOCABULARY_SIZE,
+        embedding_dim=TABLE2_EMBEDDING_SIZE,
+    )
+
+    feature1 = keras_rs.layers.FeatureConfig(
+        name="feature1",
+        table=table1,
+        input_shape=(PER_REPLICA_BATCH_SIZE,),
+        output_shape=(PER_REPLICA_BATCH_SIZE, TABLE1_EMBEDDING_SIZE),
+    )
+    feature2 = keras_rs.layers.FeatureConfig(
+        name="feature2",
+        table=table2,
+        input_shape=(PER_REPLICA_BATCH_SIZE,),
+        output_shape=(PER_REPLICA_BATCH_SIZE, TABLE2_EMBEDDING_SIZE),
+    )
+
+    feature_configs = {
+        "feature1": feature1,
+        "feature2": feature2,
+    }
+
+    embedding = keras_rs.layers.DistributedEmbedding(feature_configs)
+    ```
+
+    ## Optimizers
+
+    Each embedding table within `DistributedEmbedding` uses its own optimizer
+    for training, which is independent from the optimizer set on the model via
+    `model.compile()`.
+
+    Note that not all optimizers are supported. Currently, the following are
+    supported:
+
+    - `keras.optimizers.Adagrad`
```

Comment:
Note that JAX only currently supports Adagrad and SGD.  We default to SGD if you specify a different one.
I don't raise an error because the default "Adam" isn't supported.

Reply:
Oh, should we change the default in `TableConfig` then?

Reply:
You could do.  I don't really mind either way.  The optimizers I think usually default to Adam in keras.

Reply:
Will JAX support Adam soonish?

Reply:
Depends on the priority and what "soonish" means.  It's on the TODO list.

Reply:
Ok, I'm leaving "Adam" as the default then.